tests/cover/test_simple_characters.py

```diff
@@ -84,6 +84,23 @@ def test_find_something_rare():
         find(st, lambda c: unicodedata.category(c) != 'Zs')
 
 
+def test_whitelisted_characters_alone():
+    with pytest.raises(InvalidArgument):
+        characters(whitelist_characters=u'te02тест49st').example()
+
+
+def test_whitelisted_characters_override():
+    good_characters = u'teтестst'
+    st = characters(min_codepoint=ord('0'), max_codepoint=ord('9'),
+                    whitelist_characters=good_characters)
+
+    st.filter(lambda c: c in good_characters).example()
+    st.filter(lambda c: c in '0123456789').example()
+
+    with pytest.raises(NoSuchExample):
+        find(st, lambda c: c not in good_characters + '0123456789')
```

Comment:
If I do _just_ that, then the test fails, because the exception that is raised changes from `NoSuchExample` to `NoExample`. 

```
_________________________________ test_whitelisted_characters_override _________________________________
../../.cache/hypothesis-build-runtimes/.tox/coverage/lib/python3.6/site-packages/hypothesis/searchstrategy/strategies.py:139: in example
    verbosity=Verbosity.quiet,
../../.cache/hypothesis-build-runtimes/.tox/coverage/lib/python3.6/site-packages/hypothesis/core.py:784: in find
    runner.valid_examples,))
E   hypothesis.errors.Unsatisfiable: Unable to satisfy assumptions of lambda x: True. Only 0 examples considered satisfied assumptions

During handling of the above exception, another exception occurred:
tests/cover/test_simple_characters.py:101: in test_whitelisted_characters_override
    st.filter(lambda c: c not in good_characters + '0123456789').example()
../../.cache/hypothesis-build-runtimes/.tox/coverage/lib/python3.6/site-packages/hypothesis/searchstrategy/strategies.py:144: in example
    u'Could not find any valid examples in 100 tries'
E   hypothesis.errors.NoExamples: Could not find any valid examples in 100 tries
```
If I _also_ change the checked exception to `with pytest.raises(NoExamples)` then the test passes. Is it still a reliable test though? Would that introduce flakiness?